File: src/test/java/com/fluxchess/pulse/CastlingTest.java
/*
 * Copyright (C) 2013-2019 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */
package com.fluxchess.pulse;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class CastlingTest {

	@Test
	public void testValueOf() {
		assertThat(Castling.valueOf(Color.WHITE, CastlingType.KINGSIDE), is(Castling.WHITE_KINGSIDE));
		assertThat(Castling.valueOf(Color.WHITE, CastlingType.QUEENSIDE), is(Castling.WHITE_QUEENSIDE));
		assertThat(Castling.valueOf(Color.BLACK, CastlingType.KINGSIDE), is(Castling.BLACK_KINGSIDE));
		assertThat(Castling.valueOf(Color.BLACK, CastlingType.QUEENSIDE), is(Castling.BLACK_QUEENSIDE));
	}
}

File: src/test/cpp/castlingtest.cpp
/*
 * Copyright (C) 2013-2019 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

#include <color.h>
#include <castlingtype.h>
#include "castling.h"

#include "gtest/gtest.h"

using namespace pulse;

TEST(castlingtest, testValueOf) {
	EXPECT_EQ(+castling::WHITE_KINGSIDE, castling::valueOf(Color::WHITE, castlingtype::KINGSIDE));
	EXPECT_EQ(+castling::WHITE_QUEENSIDE, castling::valueOf(Color::WHITE, castlingtype::QUEENSIDE));
	EXPECT_EQ(+castling::BLACK_KINGSIDE, castling::valueOf(Color::BLACK, castlingtype::KINGSIDE));
	EXPECT_EQ(+castling::BLACK_QUEENSIDE, castling::valueOf(Color::BLACK, castlingtype::QUEENSIDE));
}
